aea/configurations/manager.py

```diff
@@ -381,7 +384,15 @@ def set_variable(self, path: VariablePath, value: JSON_TYPES) -> None:
             # agent
             overrides.update(data)
 
-        self.update_config(overrides)
+        dict_overrides: Optional[Dict] = None
+        if isinstance(value, (dict, OrderedDict)):
+            dict_overrides = {
+                component_id: [
+                    json_path,
+                ]
+            }
+
+        self.update_config(overrides, dict_overrides=dict_overrides)
```

Comment:
We check if the value object is a dictionary, and generate a component id to path list mapping. This will be used by the lower level methods to perform recursive methods.